File: src/main/kotlin/javaset/sets/SimpleSet.java
package javaset.sets;

import javaset.model.TestBox;
import javaset.model.TestBoxMode;

import java.util.*;

public class SimpleSet {

    public static void main(String[] args) {

        Set<Object> hashSet = new HashSet<>();
        hashSet.add(new TestBox(1, "1"));
        hashSet.add(new TestBox(1, "1"));
        hashSet.add(new TestBoxMode(1, "1"));
        hashSet.add(new TestBoxMode(1, "1"));
        hashSet.add(new TestBox(2, "2"));
        hashSet.add("100");
        hashSet.add(100);
        System.out.println("hashSet: " + hashSet);


        Set<Object> treeSet = new TreeSet();
        treeSet.add(101);
        treeSet.add(17);
        treeSet.add(57);
        treeSet.add(39);
        System.out.println("treeSet: " + treeSet);


        Set<Object> linkedHashSet = new LinkedHashSet<>();
        linkedHashSet.add(new TestBox(1, "1"));
        linkedHashSet.add(new TestBox(1, "1"));
        linkedHashSet.add(new TestBoxMode(1, "1"));
        linkedHashSet.add(new TestBoxMode(1, "1"));
        linkedHashSet.add("17");
        linkedHashSet.add(57);
        linkedHashSet.add(39);
        System.out.println("linkedHashSet: " + linkedHashSet);
    }
}

File: src/main/kotlin/kotlinset/sets/SimpleSet.kt
package kotlinset.sets

import kotlinset.model.TestBox
import kotlinset.model.TestBoxMode

fun main() {

    val simpleSet = setOf(
        TestBox(1, "1"),
        TestBox(1, "1"),
        TestBoxMode(1, "1"),
        TestBoxMode(1, "1"),
        "101",
        37
    )
    println(simpleSet)

    val treeSet = sortedSetOf(17, 7, 19, 29, 101, 5)
    println(treeSet)

    val linkedSet = linkedSetOf(
        TestBox(1, "1"),
        TestBox(1, "1"),
        TestBoxMode(1, "1"),
        TestBoxMode(1, "1"),
        "101",
        37
    )
    println(linkedSet)

    val hashSet = hashSetOf(
        TestBox(1, "1"),
        TestBox(1, "1"),
        TestBoxMode(1, "1"),
        TestBoxMode(1, "1"),
        "101",
        37
    )
    println(hashSet)
}